assets/src/scripts/store/instantaneous-value-time-series-data.js

```diff
@@ -0,0 +1,361 @@
+        } else {
+            return Promise.resolve({});
+        }
+    };
+};
+
+/*
+ * Asynchronous Redux Action which retrieves data for a custom time range
+ * @param {String} siteno
+ * @param {String} startTimeStr
+ * @param {String} endTimeStr
+ * @param {String} paramCd
+ * @return {Function} which returns a promise when the data has been fetched
+ */
+const retrieveUserRequestedIVDataForDateRange = function(siteno, startTimeStr, endTimeStr, parmCd=null) {
+    return function(dispatch, getState) {
+        const state = getState();
+        const locationIanaTimeZone = getIanaTimeZone(state);
+        const startTime = new DateTime.fromISO(startTimeStr,{zone: locationIanaTimeZone}).toMillis();
+        const endTime = new DateTime.fromISO(endTimeStr, {zone: locationIanaTimeZone}).toMillis();
+        return dispatch(Actions.retrieveCustomIVTimeSeries(siteno, startTime, endTime, parmCd));
+    };
+};
+
+/*
+* Asynchronous Redux Action which sets the current variable id and fetches the custom
+* time series for that variable or the extended time series depending on the current date range kind.
+* @param {String} siteno
+* @param {String} variableID
+* @return {Function} when returns a promise.
+ */
+const updateIVCurrentVariableAndRetrieveTimeSeries = function(siteno, variableID) {
+    return function(dispatch, getState) {
+        dispatch(ivTimeSeriesStateActions.setCurrentIVVariable(variableID));
+        const state = getState();
+        const currentDateRange = getCurrentDateRangeKind(state);
+        if (currentDateRange === 'custom') {
+            const timeRange = getCustomTimeRange(state);
+            return dispatch(
+                Actions.retrieveCustomIVTimeSeries(siteno, timeRange.start, timeRange.end));
+        } else {
+            return dispatch(Actions.retrieveExtendedIVTimeSeries(siteno, currentDateRange));
+        }
+    };
+};
+
+export const ivTimeSeriesDataReducer = function(ivTimeSeriesData={}, action) {
+    switch(action.type) {
+        case 'ADD_IV_TIME_SERIES_COLLECTION':
+            return merge({}, ivTimeSeriesData, action.collection);
+
+        case 'RESET_IV_TIME_SERIES': {
+            let newSeries = {
+                ...ivTimeSeriesData,
+                timeSeries: omitBy(ivTimeSeriesData.timeSeries, (tsValue) => tsValue.tsKey === action.tsRequestKey),
+                requests: {
+                    ...(ivTimeSeriesData || {}).requests
+                }
+            };
+            newSeries.requests[action.tsRequestKey] = {};
+
+            return newSeries;
+        }
+
+        default:
+            return ivTimeSeriesData;
+    }
+};
+
+export const Actions = {
+    addIVTimeSeriesCollection,
+    resetIVTimeSeries,
+    retrieveCompareIVTimeSeries,
+    retrieveIVTimeSeries,
+    retrieveCustomTimePeriodIVTimeSeries,
+    retrieveCustomIVTimeSeries,
+    retrieveExtendedIVTimeSeries,
+    retrieveUserRequestedIVDataForDateRange,
+    updateIVCurrentVariableAndRetrieveTimeSeries
```

Comment:
I don't know that I remember seeing this grab bag of Actions before, but I like that they are all in one spot.

Reply:
They were all buried in the store/index.js module along with the tests. I'm hoping this will make it easier ultimately to make the code reusuable across different pages (or maybe applications).